File: SocialistB/src/main/java/com/ss/SocialistB/service/FriendService.java
package com.ss.SocialistB.service;

import java.util.List;

import com.ss.SocialistB.model.Friend;
import com.ss.SocialistB.model.User;

public interface FriendService {
	List<User> listOfSuggestedUsers(String userName);
	void friendRequest(Friend friend);
	List<Friend> pendingRequests(String toID);
	void updatePendingRequests(Friend friend);
	List<String> listOfFriends(String userName);
}

File: SocialistF/WebContent/services/friendservice.js
app.factory('FriendService',function($http){
	var friendService={}
	var BASE_URL="http://localhost:8087/SocialistM"
		
		friendService.listOfSuggestedUsers = function()
		{
			return $http.get(BASE_URL+"/getSuggestedUsers")
		}
	friendService.sendFriendRequest = function(toID)
	{
		return $http.get(BASE_URL+"/friendRequest/"+toID)
	}

	friendService.pendingRequests= function()
	{
		return $http.get(BASE_URL+"/pendingRequests")
	}
	friendService.updatePendingRequest= function(request)
	{
		return $http.put(BASE_URL+"/updatePendingRequests",request)
	}
	friendService.listOfFriends=function()
	{
		return $http.get(BASE_URL+"/listOfFriends")
	}
	
	return friendService;
})